### PR TITLE
[opinfo][fix] conv3d & fix conv{1, 2}d for neg dilation|groups & add ErrorInputs for conv ops

### PR DESCRIPTION
**Pull request by khushi-411**
Previous PR: https://github.com/pytorch/pytorch/pull/85202

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

Also, cc'ing @lezcano @kshitij12345 @zou3519, who reviewed my previous PR. Thanks!

### CODE REVIEW

File: aten/src/ATen/native/Convolution.cpp [hunk 1]
```diff
@@ -972,7 +981,7 @@ static Tensor convolution_same(
     SymIntArrayRef stride, SymIntArrayRef dilation, c10::SymInt groups) {
 
   auto k = weight.dim();
-  TORCH_CHECK(k > 2, "weight should have at least three dimensions");
```
Comment (lezcano) [inline aten/src/ATen/native/Convolution.cpp:975]: why this change

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3866,33 +3862,100 @@ def sample_inputs_conv1d(op_info, device, dtype, requires_grad, **kwargs):
...
+    yield ErrorInput(
+        SampleInput(make_arg((2, 2, 4)), args=(make_arg((2, 2, 2)), make_arg((2,))),
+                    kwargs={'padding': 'same', 'groups': 0}), error_regex="non-positive groups is not supported")
 
 
 def error_inputs_conv2d(opinfo, device, **kwargs):
-    weight = torch.randint(high=10, size=(3, 2, 3, 3), device=device)
-    input = torch.randint(high=10, size=(2, 4, 4), device=device)
-    bias = torch.rand((3,), dtype=torch.float32, device=device)
-    yield ErrorInput(SampleInput(input, args=(weight, bias)), error_regex="should be the same")
```
Comment (lezcano) [inline torch/testing/_internal/common_methods_invocations.py:3882]: Why did you remove this sample (and the one below)? Where is this tested?

File: aten/src/ATen/native/Convolution.cpp (hunk 1 above)
Comment (khushi-411) [reply aten/src/ATen/native/Convolution.cpp:975]: We don't need to remove this check. I've updated it. Thanks for pointing it out, @lezcano!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:3882]: Added this and the below case, using `make_arg` in my latest commit https://github.com/pytorch/pytorch/pull/113885/commits/a358f6c06bbcd57ddc12941aeceb0a752a44f3b2

File: test/functorch/test_ops.py [hunk 3]
```diff
@@ -1808,6 +1808,10 @@ def fn(input, weight, bias):
              {torch.float32: tol(atol=2e-04, rtol=1e-04)}, device_type='cuda'),
         tol2('linalg.pinv', 'hermitian',
              {torch.float32: tol(atol=5e-06, rtol=5e-06)}),
+        tol1('nn.functional.conv3d',
+             {torch.float32: tol(atol=5e-04, rtol=9e-03)}, device_type='cuda'),
+        tol1('nn.functional.conv3d',
+             {torch.float32: tol(atol=5e-04, rtol=9e-03)}, device_type='cpu'),
```
Comment (lezcano) [inline test/functorch/test_ops.py:1814]:
```suggestion
        tol1('nn.functional.conv3d',
             {torch.float32: tol(atol=5e-04, rtol=9e-03)}),
```

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (lezcano) [reply torch/testing/_internal/common_methods_invocations.py:3882]: do it in all the conv operators please

Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:3882]: @lezcano, `conv1d`, and `conv3d` does not throw error, unlike `conv2d`. Please see:

**For conv1d:**
```python
>>> import torch
>>> input = torch.randint(high=10, size=(1, 1, 4))
>>> weight = torch.randint(high=10, size=(1, 1, 2))
>>> bias = torch.rand(1, dtype=torch.float64)
>>> torch.nn.functional.conv1d(input, weight, bias)
tensor([[[86, 97, 90]]])
>>> bias = torch.rand(1, dtype=torch.complex128)
>>> torch.nn.functional.conv1d(input, weight, bias)
<stdin>:1: UserWarning: Casting complex values to real discards the imaginary part (Triggered internally at /home/khushi/Documents/hello/pytorch/aten/src/ATen/native/Copy.cpp:301.)
tensor([[[86, 97, 90]]])
```
**For conv3d:**
```python
>>> input = torch.randint(high=10, size=(1, 1, 4, 4, 4))
>>> weight = torch.randint(high=10, size=(1, 1, 2, 2, 2))
>>> bias = torch.rand(1, dtype=torch.float64)
>>> torch.nn.functional.conv3d(input, weight, bias)
tensor([[[[[196, 192, 212],
           [155, 184, 136],
           [123, 152, 111]],

          [[161, 230, 240],
           [189, 240, 236],
           [214, 137, 105]],

          [[137, 196, 232],
           [127, 187, 293],
           [118, 206, 196]]]]])
>>> bias = torch.rand(1, dtype=torch.complex128)
>>> torch.nn.functional.conv3d(input, weight, bias)
tensor([[[[[196, 192, 212],
           [155, 184, 136],
           [123, 152, 111]],

          [[161, 230, 240],
           [189, 240, 236],
           [214, 137, 105]],

          [[137, 196, 232],
           [127, 187, 293],
           [118, 206, 196]]]]])
```
I think we should fix this, maybe in a different PR. I'll work on this. wdyt? 

Thanks!

Comment (lezcano) [reply torch/testing/_internal/common_methods_invocations.py:3882]: Sure, different PR SGTM

**Review by lezcano — APPROVED** [review]

**Comment by khushi-411** [comment]
Thanks, @lezcano, for all your helpful reviews!

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting you change but its TestCommonCUDA::test_compare_cpu_nn_functional_conv3d test failing in trunk https://hud.pytorch.org/pytorch/pytorch/commit/4fa1ff8404b6c26c076288aa2a0aa77f0c24916a' -m nosignal

The test is slow so, it wasn't run as part of regular CI.  I have added `ciflow/slow` to run it on your PR.  Please help take a look and reland the change.
